### ISSUE TITLE
A survey of loss function for semantic segmentation

### ISSUE THREAD

**Issue by RandyGen** [RandyGen/reading_papers#24]
# A survey of loss function for semantic segmentation
https://arxiv.org/abs/2006.14822

## Abst
- image segmentationの分野は多くの応用先アプリがある
    - 多くのアプリやケースに応じてモデルを最適化する必要がある
        - そこでモデルに使用されるlossをまとめた
        - log-cosh dice loss を提案

## Intro
- 深層学習は広い分野で活用されている
    - medical 分野の segmentation も同様
- segmentation 分野は多くのカテゴリに含まれるpixelを分類する
    - モデルの損失関数の選択が重要
        - 15種の損失関数を要約、4つにカテゴライズ
- 新たなsegの損失関数を提案
    - skull-strippingで検証

## LF
- DLアルゴリズムはモデルの学習のため確率的勾配降下法が使用
    - 正確かつ早い学習のために境界線を捉えた損失関数が必要

### BCE
- 2つの異なる確率分布を図るもの

### WCE
- positive exsampleにweighting
    - false positiveにturning

### BCE
- WCE+negative exampleにもweighting

### FL
- BCEの派生で、easy exampleをdown-weightし、hard exampleにfocus
    - γ=1でCE
    - αは[0,1]
        - inverse class frequency
        - hyper-parameter

### DL
- 2つの画像のsimilarityを計算
    - y=p=0のときDL=0

### Tversky loss
- Tversky index is generalization of Dice cofficient
    - add weights(β) to FP and FN
    - β=1/2 is Dc

### FTL
- FLに似て、easy exampleにdown weights
    - FLとTLを組み合わせたもの

### SSL
- Sensitive Specificity Loss
    - DcのSS版

### SL
- Shape-aware Loss
    - 境界線付近のユーグリッド距離の平均を計算

### CL
- Combo Loss
    - DLとM-BCEの加重和

### ELL
- Exponential Logarithmic Loss
    - DLとCEの加重和
        - -ln(DC)^γ

## Exp
- 10Encoder,8Decoderの2D U-Net
- NBFS Skull-stripping dataset
- 9つのloss
- Dc, Sensitivity, Specificityで評価

### Evalution
- Log-Cosh DL or FTL

## Conclusion
- datasetに強く依存
    - distribution, skewness, boundaries
    - すべてのケースに最適なlossはない
        - imbalanceの対処には効果的
        - balanceにはBCE
- few-shot でも同様の実験をしたい
